src/components/types.ts

```diff
@@ -0,0 +1,41 @@
+// ignore next line
+/// <reference path="../../node_modules/unlayer-types/embed.d.ts" />
+
+import Embed from 'embed/index';
+import { Editor as EditorClass } from 'embed/Editor';
+import { Config } from "embed/Config";
+import { AppearanceConfig, DisplayMode, ToolsConfig } from 'state/types/types';
+
+export type Unlayer = typeof Embed;
+export type Editor = InstanceType<typeof EditorClass>;
+
+export interface EmailEditorProps {
+  editor: Editor | null;
+  editorId?: string | undefined;
+  minHeight?: number | string | undefined;
+  onLoad?(unlayer: Editor): void;
+  onReady?(unlayer: Editor): void;
+  options?: Config;
+  scriptUrl?: string | undefined;
+  // style?: CSSProperties | undefined;
```

Comment:
Shouldn't we keep support for this prop? If not, then we should remove this line

Reply:
If it was a property used before for a reason, we should probably keep it for backward compatibility and set it as deprecated as below

Reply:
the style prop was carried over from react editor, it was using a react specific `CSSProperties` which is why i have removed it completely. We weren't supporting this prop in vue-email-editor before

I have also updated and removed unrelated props and added deprecated flags